src/Store.js

```diff
@@ -345,7 +346,20 @@ class Store {
     snapshotData.values.forEach(addToStream)
     rs.push(null) // tell the stream we're finished
 
-    const snapshot = this._ipfs.files.add ? await this._ipfs.files.add(rs) : await this._ipfs.add(rs)
+    // js-ipfs >= 0.41, ipfs.add doesn't accept a Readable Stream
+    const buf = rs.read(Infinity)
+
+    let snapshot = this._ipfs.files.add ? await this._ipfs.files.add(buf) : await this._ipfs.add(buf)
+
+    if (!Array.isArray(snapshot)) { // js-ipfs >= 0.41, ipfs.add returns an async iterable
```

Comment:
Will snapshot ever be an array? If so should we handle that?

Reply:
ipfs.add returns [an async iterable](https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/FILES.md#add) instead of an array since 0.41.0